src/de/jost_net/JVerein/io/AbrechnungSEPA.java

```diff
@@ -781,11 +782,11 @@ private void abbuchenZusatzbetraege(AbrechnungSEPAParam param,
           zahler.setZahlungsweg(new Zahlungsweg(zahlungsweg));
           if (z.getBuchungsart() != null)
           {
-            zahler.setBuchungsartId(z.getBuchungsart().getID());
+            zahler.setBuchungsartId(Long.parseLong(z.getBuchungsart().getID()));
```

Comment:
Wäre es nicht performanter wenn man in Zusatzbetrag eine Methode getbuchungsartId() einbauen würde, dann bräuchte man kein Buchungsklasse Objekt erzeugen und die Id parsen.

Reply:
ja, das ist besser. habe ich hinzugefügt, auch bei der Beitragsgruppe

Reply:
Bei Beitragsgruppe funktioniert das so aber nicht. Hier ist das Attribut kein Long sondern Buchungsart. Das ist in getForeignObject so definiert.
Ich habe auch lange gebraucht bis ich das verstanden habe.
Du müsstest die Buchungsart aus dieser Methode raus löschen und dann aber die getBuchungsart() Methode anpassen so dass sie auch das Long holt und die Buchungsart aus dem Cache so wie bei den anderen Klassen.

Reply:
Habe noch was vergessen. Bei der Methode getAttribute() muss man die buchungsart auch aufnehmen und die getter Methode aufrufen weil das Attribut ja jetzt ein Long ist und bei getAttribute die Klasse zurückgeben werden muss.

Reply:
klar, hatte ich vergessen